src/main/java/com/ays/auth/model/AysIdentity.java

```diff
@@ -36,4 +40,14 @@ private Jwt getJwt() {
         return ((Jwt) SecurityContextHolder.getContext().getAuthentication().getPrincipal());
     }
 
+
+    private Authentication getAuthentication() {
+        return SecurityContextHolder.getContext().getAuthentication();
+    }
+
+    public List<String> getUsersType() {
+        Authentication authentication = getAuthentication();
+        return authentication.getAuthorities().stream().map(GrantedAuthority::getAuthority).toList();
+    }
```

Comment:
If we remove the two methods here and use the following method, it will work properly. Additionally, let's not forget to add JavaDoc.

```
public AysUserType getUserType() {
   return AysUserType.valueOf(this.getJwt().getClaim(AysTokenClaims.USER_TYPE.getValue()));
}
```

Reply:
I just revised the code. Can you look through it if you have enough free time?